Completion/Unix/Command/_git

```diff
@@ -1567,13 +1567,14 @@ _git-restore() {
 
   case $state in
     pathspecs)
-      if [[ -z ${opt_args[(I)-s|--source|-S|--staged]} ]] &&
-	# use index as a default base unless -S is specified
-	__git_ignore_line __git_modified_files
+      if [[ -z ${opt_args[(I)-S|--staged]} ]]
       then
-	ret=0
+        __git_ignore_line __git_changed-in-working-tree_files
+      elif [[ -z ${opt_args[(I)-W|--worktree]} ]]
+      then
+        __git_ignore_line __git_changed-in-index_files
```

Comment:
Finally got back to this.  Sorry for the long delay.

I'm reviewing this against git-restore(1) from git 1:2.25.1-1 (Debian package) and have a few concerns:

1. The man page says the command is experimental and its behaviour may change.  This would seem to imply that the command's current behaviour is not guaranteed to be forward compatible with its future behaviour.  How should `_git-restore` handle this?  Perhaps it should default to doing nothing unless the user has opted in to completing experimental commands?  (The opt in will likely take the form of a zstyle setting.)

    This is not a regression, however, and as such not a blocker.

2. The `if` and `elif` branch should have `&& ret=0` appended to them.

3. It's possible to specify both -S and -W.  In that case, I assume that files modified _either_ in the index or in the worktree should be offered.  I think `__git_changed_files` will be the right function to use in this case.

Putting this all together, I think the following would be better:

    if --staged and --worktree:
        __git_ignore_line __git_changed_files && ret=0
    elif --staged:
        __git_ignore_line __git_changed-in-index_files && ret=0
    elif --worktree:
        __git_ignore_line __git_changed-in-working-tree_files && ret=0
    else:
        __git_ignore_line __git_tree_files ${PREFIX:-.} ${(Qv)opt_args[(i)-s|--source]:-HEAD} && ret=0

WDYT?

Apologies again for the delay.